### PR TITLE
FOGL-8193 Add OMFBuffer class

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/plugins/north/OMF/omf.cpp [hunk 1]
```diff
@@ -1525,13 +1527,14 @@ uint32_t OMF::sendToServer(const vector<Reading *>& readings,
 								   timeT3,
 								   timeT4,
 								   readings.size(),
-								   json_not_compressed.length(),
-								   json.length()
+								   paylaod.size(),
```
Comment (praveen-garg) [inline C/plugins/north/OMF/omf.cpp:1530]: typo

**Review by praveen-garg — COMMENTED** [review]
Please update with develop, and also check new unit tests failure.

Compile & run with Instrumentation enabled, as well, for next commit.

File: C/plugins/north/OMF/omf.cpp [hunk 2]
```diff
@@ -1525,13 +1530,14 @@ uint32_t OMF::sendToServer(const vector<Reading *>& readings,
 								   timeT3,
 								   timeT4,
 								   readings.size(),
-								   json_not_compressed.length(),
-								   json.length()
+								   payload.size(),
+								   payload.size()
```
[the remark this answers is not in the document]
Comment (praveen-garg) [reply C/plugins/north/OMF/omf.cpp:1534]: Compile with `#define INSTRUMENTATION 1`

File: C/plugins/north/OMF/omfbuffer.cpp [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+ * Construct a buffer with a standard size initial buffer.
+ */
+OMFBuffer::Buffer::Buffer() : offset(0), length(BUFFER_CHUNK), attached(true)
+{
+	data = new char[BUFFER_CHUNK+1];
+	data[0] = 0;
+}
+
+/**
+ * Construct a large buffer, passing the size of buffer required. THis is useful
```
Comment (RVerhoeff) [inline C/plugins/north/OMF/omfbuffer.cpp:308]: Typo: THis -> This. Very minor.

**Review by RVerhoeff — APPROVED** [review]

**Review by pintomax — DISMISSED** [review]

**Review by RVerhoeff — APPROVED** [review]

**Review by pintomax — APPROVED** [review]

File: C/plugins/north/OMF/linkdata.cpp [hunk 4]
```diff
@@ -62,17 +62,17 @@ static std::string DataPointNamesAsString(const Reading& reading)
 /**
  * OMFLinkedData constructor, generates the OMF message containing the data
  *
+ * @param payload	    The buffer into which to populate the payload
  * @param reading           Reading for which the OMF message must be generated
  * @param AFHierarchyPrefix Unused at the current stage
  * @param hints             OMF hints for the specific reading for changing the behaviour of the operation
+ * @param delin		    Add a delimiter before outputtign anything
```
Comment (RVerhoeff) [inline C/plugins/north/OMF/linkdata.cpp:69]: Typo: outputtign -> outputting

**Review by RVerhoeff — APPROVED** [review]
There is one typo but it is in a documentation header

**Review by RVerhoeff — APPROVED** [review]

**Pull request merged**